app/src/main/java/com/github/sdpteam15/polyevents/view/activity/staff/StaffRequestsActivity.kt

```diff
@@ -0,0 +1,337 @@
+            if(it.sender != currentDatabase){
+                currentDatabase.itemDatabase!!.updateItem(it.value.first,it.value.second,it.value.third)
+            }
+        }
+
+        items.group(this) { it.first.itemId!! }.then.map(this, itemNames) {
+            it[0].first.itemName!!
+        }
+
+        requests.group(this, materialRequest) {
+            staffRequestStatusFromMaterialStatus(it.status)
+        }
+
+
+        users.group(this) { it.uid }.then.map(this, userNames) { it[0].name ?: "UNKNOWN" }
+
+
+    }
+
+    override fun onResume() {
+        super.onResume()
+        getItemRequestsFromDB()
+    }
+
+    /**
+     * Gets the item request of the user and then gets the item list
+     */
+    private fun getItemRequestsFromDB() {
+        //Gets the item request of the user and then gets the item list
+
+        currentDatabase.materialRequestDatabase!!.getMaterialRequestList(
+            requests,
+            {
+                it.orderBy(MATERIAL_REQUEST_STATUS.value)
+            }
+        ).observeOnce(this) {
+            if (!it.value) {
+                showToast("Failed to get the list of material requests", this)
+            } else {
+                val sentEventIds = mutableListOf<String>()
+                for (request in requests) {
+                    if (request.eventId !in sentEventIds) {
+                        sentEventIds.add(request.eventId)
+                        val event = Observable<Event>()
+                        val zone = Observable<Zone>()
+                        currentDatabase.eventDatabase!!.getEventFromId(request.eventId, event)
+                            .observeOnce(this) {
+                                if (it.value) {
+                                    currentDatabase.zoneDatabase!!.getZoneInformation(
+                                        event.value!!.zoneId!!,
+                                        zone
+                                    ).observeOnce(this) {
+                                        if (it.value) {
+                                            zoneNameFromEventId[event.value!!.eventId!!] =
+                                                zone.value!!.zoneName!!
+                                        }
+                                    }
+                                }
+                            }
+                    }
+
+                }
+            }
+        }
+
+        currentDatabase.itemDatabase!!.getItemsList(items)
+            .observeOnce(this) {
+                if (!it.value) {
+                    showToast("Failed to get the list of items", this)
+                }
+            }
+
+
+        currentDatabase.userDatabase!!.getListAllUsers(users)
+            .observeOnce(this) {
+                if (!it.value) {
+                    showToast("Failed to get the list of users", this)
+                }
+            }
+
+        val tempZones = ObservableList<Zone>()
+        tempZones.group { }
```

Comment:
What does it do? How does it group them?

Reply:
Thanks, I forgot to remove this